_profiles/drupal/resource-versioning/index.md

```diff
@@ -0,0 +1,273 @@
+    context resource object.
+  - `successor-version`: links to the version which immediately succeeded the
+    context resource object.
+  - `prior-working-copy`: links to the working copy which immediately preceded
+    the context resource object.
+  - `subsequent-working-copy`: links to the working copy which immediately
+    preceded the context resource object.
+
+A server **MAY** provide an array of any of these links to support branching.
+
+For example, in the following version history:
+
+```
+    _c_
+   /   \
+  /__b__d__   __f__   __h
+ /         \ /     \ /
+a           e       g
+```
+
+`g` is the latest version. Both `a` and `e` were previously the latest version.
+No other revisions were ever the latest version. In this example, the following
+links could be provided:
+
+| Revision | `latest-version` | `working-copy` | `predecessor-version` | `successor-version` | `prior-working-copy` | `subsequent-working-copy` |
+| :------: | :--------------: | :------------: | :-------------------: | :-----------------: | :------------------: | :-----------------------: |
+| `a`      | `g`              | `h`            | no link               | `e`                 | no link              | `b`, `c`                  |
+| `b`      | `g`              | `h`            | `a`                   | `e`                 | `a`                  | `c`                       |
+| `c`      | `g`              | `h`            | `a`                   | `e`                 | `b`                  | `d`                       |
+| `d`      | `g`              | `h`            | `a`                   | `e`                 | `b`, `c`             | `e`                       |
+| `e`      | `g`              | `h`            | `a`                   | `g`                 | `d`                  | `f`                       |
+| `f`      | `g`              | `h`            | `e`                   | `g`                 | `e`                  | `g`                       |
+| `g`      | no link          | `h`            | `e`                   | no link             | `f`                  | `h`                       |
+| `h`      | `g`              | no link        | `g`                   | no link             | `g`                  | no link                   |
+
+> Note: In this example, `f` has both a `predecessor-version` and
+> `prior-working-copy` link to `e` because `e` was a version and `e` was also the
+> revision to which changes could be applied prior to `f`'s creation.
+
+# Version History
+
+A server **MAY** provide a "version history" endpoint. The primary data
+of a response document from a version history endpoint must be a collection of
+resource objects.
+
+Unless an `id` contains version information, the `type` and `id` members of each
+resource object in the collection **MUST** be the same.
+
+If provided, resource objects' `self` links **MUST NOT** be the same.
+
+# Version Negotiators
```

Comment:
I ❤️ the idea of having these basic version negotiators in the spec.

In addition to that we should add links to point to these negotiator definitions. That will allow servers that implement custom version negotiators to surface documentation about those. Servers implementing `id` and `rel` can point to this very page, custom ones will have to live elsewhere.

Maybe we should have links at the top level document like `"version-negotiator:rel": "http:/…/#relative-version-negotiator"` if the `resourceVersion` query string parameter is present.

Reply:
I _really_ like the ID of making negotiator IDs into URIs ❤️ good idea!

Reply:
Let's continue this conversation in this comment https://github.com/json-api/json-api/pull/1333#discussion_r243464998